### PR TITLE
worker/: don't quit worker immediately when dialed master is offline

### PR DESCRIPTION
**Pull request by lichunzhu**
### What problem does this PR solve? 
https://github.com/pingcap/dm/issues/560
When worker dials an offline master, it will returns error directly and stop the worker process.

### What is changed and how it works?
1. add `WithBlock` option for grpc dial to avoid dialing an offline master.
2. don't return error immediately when dialed master is offline. Try to dial other endpoint in `cfg.Join`.
3. do the register request in the loop to avoid dialing an online master which is blocked from other dm-masters.

### Check List 

Tests 

 - Unit test
 - Integration test

### CODE REVIEW

File: dm/worker/join.go [hunk 1]
```diff
@@ -36,32 +36,38 @@ func GetJoinURLs(addrs string) []string {
...
+		if err != nil {
+			log.L().Error("fail to dial dm-master", zap.Error(err))
+			continue
+		}
+		client = pb.NewMasterClient(conn)
+		ctx1, cancel1 = context.WithTimeout(ctx, 3*time.Second)
+		resp, err := client.RegisterWorker(ctx1, req)
+		cancel1()
+		if err != nil {
+			log.L().Error("fail to register worker: %s", zap.Error(err))
```
Comment (csuzhangxc) [inline dm/worker/join.go]:
```suggestion
			log.L().Error("fail to register worker", zap.Error(err))
```

ditto for L65.

please also do this in `cmd/dm-worker/main.go` for `join the cluster meet error %v`.

File: dm/worker/join.go [hunk 2]
```diff
@@ -36,32 +36,38 @@ func GetJoinURLs(addrs string) []string {
...
+			continue
+		}
+		if !resp.GetResult() {
+			log.L().Error("fail to register worker", zap.String("error", resp.Msg))
+			continue
+		}
+		return nil
 	}
-	return nil
+	return errors.Errorf("cannot connect with master endpoints: %v", endpoints)
```
Comment (WangXiangUSTC) [inline dm/worker/join.go]: how about add a TODO: use terror

File: dm/worker/join.go [hunk 3]
```diff
@@ -36,32 +36,38 @@ func GetJoinURLs(addrs string) []string {
...
 		Address: s.cfg.AdvertiseAddr,
 	}
-	resp, err := client.RegisterWorker(ctx, req)
-	if err != nil {
-		return err
-	}
-	if !resp.GetResult() {
-		return errors.Errorf("fail to register worker: %s", resp.GetMsg())
+
+	var client pb.MasterClient
```
Comment (WangXiangUSTC) [inline dm/worker/join.go]: seems that we don't need to define here

**Pull request merged**